backend/lib/edgehog/containers/manual_actions/update_status.ex

```diff
@@ -0,0 +1,138 @@
+#
+# This file is part of Edgehog.
+#
+# Copyright 2024 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+defmodule Edgehog.Containers.ManualActions.UpdateStatus do
+  @moduledoc false
+  use Ash.Resource.ManualUpdate
+
+  alias Edgehog.Devices
+
+  @impl Ash.Resource.ManualUpdate
+  def update(changeset, _opts, context) do
+    changeset = check_images(changeset, context)
+    Ash.update(changeset)
+  end
+
+  defp check_images(changeset, context) do
+    %{tenant: tenant} = context
+    deployment = changeset.data
+
+    with {:ok, deployment} <-
+           Ash.load(deployment, device: [], release: [containers: [:image]], reuse_values?: true) do
+      changeset =
+        Ash.Changeset.for_update(changeset, :set_status, %{status: :pulling_images}, tenant: tenant)
+
+      images = Enum.map(deployment.release.containers, & &1.image)
+
+      pulled_images_ids =
+        deployment.device
+        |> Devices.available_images(tenant: tenant)
+        |> Enum.filter(& &1.pulled)
+        |> Enum.map(& &1.id)
+
+      missing_images = Enum.reject(images, fn image -> image.id not in pulled_images_ids end)
+
+      if missing_images == [] do
+        check_networks(changeset, context)
+      else
+        changeset
+      end
```

Comment:
1. As a rule of thumb, do not do double negations (`reject` and `not in`)
2. Reason being, I think you only need one negation here (you want to take (ie `filter`) all images `not in` pulled images)

```elixir
iex(1)> images = [1,2,3,4,5]
[1, 2, 3, 4, 5]
iex(2)> pulled = [2,4,1]
[2, 4, 1]
iex(3)> images |> Enum.reject(& &1 not in pulled)
[1, 2, 4]
```

3. Bonus, but you're treating lists like sets: you can use sets
```suggestion
      deployment_images = deployment.release.containers |> Enum.map(& &1.image.id) |> MapSet.new()

      available_images =
        deployment.device
        |> Devices.available_images(tenant: tenant)
        |> Enum.filter(& &1.pulled)
        |> Enum.map(& &1.id)
        |> MapSet.new()
        
      all_images_available? = MapSet.subset?(deployment_images, available_images)

      if all_images_available? do
        changeset
      else
        check_networks(changeset, context)
      end
```